pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -171,6 +172,20 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration, featur
 	if (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) && !localFeatureGate.Enabled(features.GracefulNodeShutdown) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: specifying shutdownGracePeriod or shutdownGracePeriodCriticalPods requires feature gate GracefulNodeShutdown"))
 	}
+
+	if len(kc.ResolverConfig) > 0 {
+		_, err := os.Open(kc.ResolverConfig)
+		if err != nil {
+			if os.IsNotExist(err) {
+				allErrors = append(allErrors, fmt.Errorf("invalid configuration: resolverConfig file %s not exist", kc.ResolverConfig))
+			} else if os.IsPermission(err) {
+				allErrors = append(allErrors, fmt.Errorf("invalid configuration: read resolverConfig file %s without permission", kc.ResolverConfig))
+			} else {
+				allErrors = append(allErrors, fmt.Errorf("invalid configuration: unknown error occurred while reading resolverConfig file %s", kc.ResolverConfig))
```

Comment:
The error is not unknown. It's available from err, I believe.
BTW, "file doesn't exist" or "permission denied" errors are also there. I don't think it makes sense to handle them separately.